### ISSUE TITLE
Cannot query field "video_id" on type "PrismicEmbedType"

### ISSUE THREAD

**Issue by rh-silpion** [angeloashmore/gatsby-source-prismic#286]
Hello,

I'm using gatsby-source-prismic version `3.2.1` and get this error for video fields:
> Cannot query field "video_id" on type "PrismicEmbedType". Did you mean "media_id"?  graphql/template-strings

From the prismic GraphQL API I get this JSON response for the video field:
```json
"video": {
  "is_plus": "1",
  "account_type": "plus",
  "width": 426,
  "height": 240,
  "duration": 42,
  "description": "",
  "thumbnail_url_with_play_button": "https://i.vimeocdn.com/filter/overlay...",
  "upload_date": "...",
  "video_id": 1111,
  "uri": "/videos/1111",
  "embed_url": "https://vimeo.com/1111",
  "type": "video",
  "version": "1.0",
  "title": "...",
  "author_name": "...",
  "author_url": "https://vimeo.com/1111",
  "provider_name": "Vimeo",
  "provider_url": "https://vimeo.com/",
  "cache_age": null,
  "thumbnail_url": "https://i.vimeocdn.com/video/...",
  "thumbnail_width": 295,
  "thumbnail_height": 166,
  "html": "<iframe src=\"https://player.vimeo.com/video/...\" width=\"426\" height=\"240\" frameborder=\"0\" allow=\"autoplay; fullscreen\" allowfullscreen title=\"...\"></iframe>"
}
```

There is no field named `media_id`, but `video_id ` exists.

**Comment by MarcMcIntosh** [comment]
Hi @rh-silpion 
what's the response from the graphql-playground?
(http://localhost:8000/__graphql)

**Comment by MarcMcIntosh** [comment]
Actually, I checked oembed response from viemo, and yes they do conatina a video_id property.

**Comment by MarcMcIntosh** [comment]
HI @rh-silpion  

Can I ask for what specific reason you need the video_id for?
I've made a pr for adding the field, but as the field isn't part of the oembed spec, it may show up as a field from providers who don't add a video_id field (like youtube)

**Comment by rh-silpion** [comment]
Hi @MarcMcIntosh

I'm no longer involved in the project where this is used. But I remenber, that we need the `video_id` for a Vimeo iframe solution like this:

```javascript
<iframe
    src={`https://player.vimeo.com/video/${props.video_id}?app_id=123;&loop=1&autoplay=1&muted=1&controls=false`}
    frameBorder="0"
    allowFullScreen
    autoPlay
    title={props.embed_title}
/>
```

From prismic.io I get the JSON from my [first post](https://github.com/angeloashmore/gatsby-source-prismic/issues/286#issue-736322706) for an [Embed field](https://user-guides.prismic.io/en/articles/484832-embed) with a Vimeo url like this: `https://vimeo.com/{ID}`

I'm not sure what you want to see in the local graphql playground. For `media_id` I get
```json
"media_id": null
```
and `video_id` didn't exists.

I fixed that with
```javascript
exports.createSchemaCustomization = ({ actions }) => {
  const { createTypes } = actions
  const typeDefs = `
    type PrismicEmbedType implements Node {
      video_id: ID
    }
  `
  createTypes(typeDefs)
}
```

and everything then worked as expected.

**Comment by angeloashmore** [comment]
Thanks @rh-silpion and @MarcMcIntosh, sorry this slipped through the cracks.

@MarcMcIntosh's PR has been merged and published in `v3.3.4` so you should be able to query the `video_id` field in Embed fields without a custom type override..

In the next major version of the plugin, Embed fields will be inferred by Gatsby rather than defined by the plugin. This means when other providers provide fields specific to their platform, like the `video_id` field, they will automatically be available.
